example/lib/providers/ditto_provider.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'dart:async';
+
+import 'package:ditto_live/ditto_live.dart';
+import 'package:permission_handler/permission_handler.dart';
+import 'package:path_provider/path_provider.dart';
+
+class DittoProvider {
+  Ditto? _ditto;
+
+  /// The Ditto instance used for database operations
+  Ditto? get ditto => _ditto;
+
+  /// Initializes the Ditto instance with necessary permissions and configuration.
+  /// https://docs.ditto.live/sdk/latest/install-guides/flutter#step-3-import-and-initialize-the-ditto-sdk
+  ///
+  /// This function:
+  /// 1. Requests required Bluetooth and WiFi permissions on non-web platforms
+  /// 2. Initializes the Ditto SDK
+  /// 3. Sets up online playground identity with the provided app ID and token
+  /// 4. Enables peer-to-peer communication on non-web platforms
+  /// 5. Configures WebSocket connection to Ditto cloud
+  /// 6. Starts sync and updates the app state with the configured Ditto instance
+  Future<void> initialize(
+    String appId,
+    String token,
+    String authUrl,
+    String websocketUrl,
+  ) async {
+    // Note: macOS handles Bluetooth permissions differently via entitlements
+    if (!kIsWeb &&  (Platform.isIOS || Platform.isAndroid)) {
+      // Request permissions and check their status
+        await [
+          Permission.bluetoothConnect,
+          Permission.bluetoothAdvertise,
+          Permission.nearbyWifiDevices,
+          Permission.bluetoothScan
+        ].request();
+      }
+    // Initialize Ditto first - this must be called before any other Ditto operations
+    await Ditto.init();
+
+    // Now we can safely configure logging
+    DittoLogger.isEnabled = false;
+    DittoLogger.minimumLogLevel = LogLevel.error;
+    DittoLogger.customLogCallback = (level, message) {
+      print("[$level] => $message");
+    };
+
+    final persistenceDirectory = await getApplicationDocumentsDirectory();
```

Comment:
We also need to remove this - the SDK handles this internally as long as the string passed in is a relative path.

What you've written is equivalent to not passing a persistence directory at all, but will fail on web.